integration-tests/actor-tests/lib/actor.ts

```diff
@@ -0,0 +1,289 @@
+import { Mutex } from 'async-mutex'
+import { sleep } from '../../test/shared/utils'
+import {
+  sanitizeForMetrics,
+  benchDurationsSummary,
+  successfulBenchRunsTotal,
+  failedActorRunsTotal,
+  successfulActorRunsTotal,
+  failedBenchRunsTotal,
+} from './metrics'
+import { ActorLogger, WorkerLogger } from './logger'
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const asyncNoop = async () => {}
+
+export type AsyncCB = () => Promise<void>
+
+export interface Bencher {
+  bench: (name: string, cb: () => Promise<void>) => Promise<void>
+}
+
+export type RunCB<C> = (b: Bencher, ctx: C) => Promise<void>
+
+export interface RunOpts {
+  runs: number | null
+  runFor: number | null
+  concurrency: number
+  thinkTime: number
+}
+
+class Latch {
+  private n: number
+
+  private p: Promise<void>
+
+  private resolver: () => void
+
+  constructor(n: number) {
+    this.n = n
+    this.p = new Promise((resolve) => {
+      this.resolver = resolve
+    })
+  }
+
+  countDown() {
+    this.n--
+    if (this.n === 0) {
+      this.resolver()
+    }
+  }
+
+  wait() {
+    return this.p
+  }
+}
+
+export class Runner {
+  private readonly workerId: number
+
+  private readonly actor: Actor
+
+  private readonly mtx: Mutex
+
+  private readonly readyLatch: Latch
+
+  private readonly stepper: Bencher
+
+  private readonly logger: WorkerLogger
+
+  constructor(workerId: number, actor: Actor, mtx: Mutex, readyLatch: Latch) {
+    this.workerId = workerId
+    this.actor = actor
+    this.mtx = mtx
+    this.readyLatch = readyLatch
+    this.stepper = {
+      bench: this.bench,
+    }
+    this.logger = new WorkerLogger(this.actor.name, workerId)
+  }
+
+  bench = async (name: string, cb: () => Promise<void>) => {
+    const metricLabels = {
+      actor_name: sanitizeForMetrics(this.actor.name),
+      bench_name: sanitizeForMetrics(name),
+    }
+    const start = performance.now()
```

Comment:
Getting an error when trying to run the tests saying that `performance` is not defined

Reply:
Ah, looks like `performance` is a global in Node 16+, but requires a package import before. I'll fix this.